src/scripts/main.js

```diff
@@ -1,5 +1,36 @@
 'use strict';
 
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
 document.addEventListener('click', (e) => {
   // write code here
+  const wallCoords = wall.getBoundingClientRect();
+  const spiderCoords = spider.getBoundingClientRect();
+  const border = parseInt(
+    window.getComputedStyle(wall).borderWidth.split('px').join(''),
+  );
+
+  let spiderY =
+    e.clientY - wallCoords.top - parseInt(spider.offsetWidth) / 2 - border;
+  let spiderX =
+    e.clientX - wallCoords.left - parseInt(spider.offsetWidth) / 2 - border;
```

Comment:
Ensure that `spider.offsetWidth` is correctly used for both `spiderY` and `spiderX` calculations. If the spider's width and height are different, you might need to use `spider.offsetHeight` for `spiderY`.